cmd/gateclient/oauth.go

```diff
@@ -0,0 +1,85 @@
+package gateclient
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os/exec"
+	"runtime"
+
+	"golang.org/x/oauth2"
+)
+
+const localWebServer = "localhost:8085"
+
+func startWebServer() (codeCh chan string, err error) {
+	listener, err := net.Listen("tcp", localWebServer)
+	if err != nil {
+		return nil, err
+	}
+	codeCh = make(chan string)
+
+	go http.Serve(listener, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// TODO: add handle of `error`
```

Comment:
This is commented but new in this PR. Can you give an example of what `error` is here?
Could be something for a follow up PR.